File: 1232-check-if-it-is-a-straight-line/1232-check-if-it-is-a-straight-line.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        (x1, y1), (x2, y2) = coordinates[:2]
        for i in range(2, len(coordinates)):
            (x, y) = coordinates[i]
            if((y2 - y1) * (x1 - x) != (y1 - y) * (x2 - x1)):
                return False
        return True
        
        #-------------------------46 / 79 test cases passed -----------------------------
        # d = abs(coordinates[0][0] - coordinates[0][1])
        # for i in coordinates:
        #     if abs(i[0]-i[1]) != d: return False
        # return True

File: 1232-check-if-it-is-a-straight-line/1232-check-if-it-is-a-straight-line.cpp
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        int x1,x2,y1,y2;
        bool val = true;
        x1 = coordinates[0][0], y1 = coordinates[0][1];
        x2 = coordinates[1][0], y2 = coordinates[1][1];
        for(int i=2;i<coordinates.size();i++){
            int x=coordinates[i][0], y=coordinates[i][1];
            if((y2-y1)*(x1-x)!=(y1-y)*(x2-x1))
                val = false;
        }
        return val;
    }
};